src/data/markdown/docs/02 javascript api/07 k6-experimental/04 tracing/01 instrumentHTTP.md

```diff
@@ -0,0 +1,51 @@
+---
+title: 'instrumentHTTP'
+excerpt: 'instrumentHTTP instruments the k6 http module with tracing capabilities.'
+---
+
+The `instrumentHTTP` function instruments the k6 http module with tracing capabilities. It transparently replaces each of the k6 http module functions with versions that automatically attach a trace context to every request. Functions that you can instrument include [del](/javascript-api/k6-http/del), [get](/javascript-api/k6-http/get), [head](/javascript-api/k6-http/head), [options](/javascript-api/k6-http/options), [patch](/javascript-api/k6-http/patch), [post](/javascript-api/k6-http/post), [put](/javascript-api/k6-http/head), and [request](/javascript-api/k6-http/request).
+
+The `instrumentHTTP` automatically adds tracing information to HTTP requests.
+This means that, to instrument the HTTP requests, you don't need to rewrite any code.
+Instead, call it once in the init context.
+From that point forward, all requests made by the HTTP module from that point forward will have a trace context header added to them, and the metadata for the data-point output will have the used `trace_id`. For details about propagation, refer to [About trace contexts](/javascript-api/k6-experimental/tracing#about-trace-contexts).
+
+The `instrumentHTTP` function takes a single configuration object argument that can be used to configure the tracer. The argument should be an instance of the [`Options`](/javascript-api/k6-experimental/tracing/options) object.
+
+## Example
+
+This example demonstrates how calling the `instrumentHTTP` function in the init context of a script once ensures that all requests made by the HTTP module from that point forward will have a trace context header attached.
+
+<CodeGroup labels={["example-tracing-instrumentHTTP.js"]} lineNumbers={[]} showCopyButton={[true]}>
+
+```javascript
+import { check } from "k6";
+import tracing from "k6/experimental/tracing";
+import http from "k6/http";
+
+// instrumentHTTP will ensure that all requests made by the http module
+// will be traced. The first argument is a configuration object that
+// can be used to configure the tracer.
+//
+// Currently supported HTTP methods are: get, post, put, patch, head,
+// del, options, and request.
+tracing.instrumentHTTP({
+  // propagator defines the trace context propagation format.
+  // Currently supported: w3c and jaeger.
+  // Default: w3c
+  propagator: "w3c",
+});
+
+export default () => {
+  const res = http.get("http://httpbin.org/get", {
+    headers: {
+      "X-Example-Header": "instrumented/get",
+    },
+  });
+  check(res, {
+    "status is 200": (r) => r.status === 200,
+  });
```

Comment:
Same comment - I think we are over adding `check`s everywhere and if we are going to be doing that, let's check for something relevant for the example :shrug: